### ISSUE TITLE
Upgrade Ark to Velero aka. make Backup test work with mtls

### ISSUE THREAD

**Issue by sayanh** [kyma-project/kyma#3760]
**Description**

Upgrade Ark to Velero v0.11.0

**Reasons**
- With current Ark, backup/restore tests seem to break the service-mesh after enabling mTLS.
- Backup/restore of ns: istio-system doesn't break the service mesh on Kyma installation with mTLS and Velero 0.11.0 from master at the mTLS merge commit: `a0c74c84ea6645ea32162f43d3bbd68517f1e5f2`

**Comment by sayanh** [comment]
> As per @Tomasz-Smelcerz-SAP 
@here
Hi! I have performed update/restore without issues in the following setup:
- velero v0.11.0
- Kyma installation with mTLS from master at the mTLS merge commit: a0c74c84ea6645ea32162f43d3bbd68517f1e5f2
- GKE cluster (edited) 
After Kyma installed, I have created two services: one with sidecar, the other without. I have also created a test Pod to test mTLS. Communication works as expected, I needed a DestinationRule to communicate with service without sidecar, communication with other service was working out-of-the-box. (edited) 
Then I backed-up and restored istio-system namespace without issues (there were some warnings)
After istio restore, I tested my services again, was working.
I also tested some other services from Kyma (like monitoring and nats) and it was working.
I have created second “copy” of my services in a new namespace (to test if Istio still injects sidecars properly) and it was also working.
Warnings from velero (perhaps this is the reason it is working):
                   ```not restored: pods "istio-sidecar-injector-7b47cb4689-sbkhn" already exists and is different from backed up version.
                   not restored: pods "istio-telemetry-bdfdf74d-jw4qs" already exists and is different from backed up version.
                   not restored: horizontalpodautoscalers.autoscaling "istio-policy" already exists and is different from backed up version.
                   not restored: services "istio-citadel" already exists and is different from backed up version.
                   not restored: services "istio-egressgateway" already exists and is different from backed up version.```
TLDR: I wasn’t able to reproduce any problem with backup/restore and mTLS.

**Comment by sayanh** [comment]
Following experiments have been tried already:
- Upgrade of Ark to Velero. [PR](https://github.com/kyma-project/kyma/pull/3840)
- Change api references of Ark to Velero. [PR](https://github.com/kyma-project/kyma/pull/3783)
  - In the same PR, other tests are commented out.
  - In the same PR, Ark-plugins component has been migrated to Velero-plugins.
- Velero-plugins is a new component which needs a new Prow plan. [PR](https://github.com/kyma-project/test-infra/pull/973)
- Lastly, PR to install Velero for backup/restore e2etest cluster is [here](https://github.com/kyma-project/test-infra/pull/972/).  


Following are the

**Comment by sayanh** [comment]
Passed tests in backup/restore test suite with velero:


- [x] myStatefulSetTest

- [x] myDeploymentTest

- [ ] myFunctionTest
- [ ] myPrometheusTest -workaround available

- [x] myGrafanaTest 
- [ ] helmBrokerTest
- [x] appBrokerTest
- [x] scAddonsTest
------ yet to test
- [ ] myNamespaceControllerTest

- [ ] apiControllerTest

- [ ] myMicroFrontendTest

- [ ] myAssetStoreTest

- [ ] myCmsTest

**Comment by sayanh** [comment]
Right now, in order to get this story done following issues need to be closed:
- [ ] Fix Function and Prometheus [tests](https://github.com/kyma-project/kyma/issues/4039)
- [ ] Fix HelmBroker [test](https://github.com/kyma-project/kyma/issues/4049)

**Comment by polskikiel** [comment]
`Helm Broker` and `App Broker` tests are failing because Secrets created by the ServiceBinding haven't OwnerReferences to the ServiceBinding.

The error which fails our tests can be found in the restored ServiceBinding spec and it looks as follows: `"reason\": \"ErrorInjectingBindResult\",\n \"message\": \"Error injecting bind result: Secret \\\"appbrokertest-e0193443-cf96-482c-bfed-bbd75ec6e858/app-binding\\\" is not owned by ServiceBinding, controllerRef: nil\"`.

What's more, we have a velero plugin which should fix that issue and set a proper ownerReferences: `kyma/tools/velero-plugins/internal/plugins/owner_reference.go`. Something can be wrong with this plugin because it worked fine when we were using ARK.

Logs from the Velero startup looks fine: 
```
time="2019-06-17T09:30:05Z" level=info msg="setting log-level to INFO"
time="2019-06-17T09:30:05Z" level=info msg="Starting Velero server v0.11.0 (ab2fc65c02ecdcb9bb1690e3afe67a3b3196aebc)" logSource="pkg/cmd/server/server.go:134"
time="2019-06-17T09:30:05Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/registry.go:98" name=pod
time="2019-06-17T09:30:05Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/registry.go:98" name=pv
time="2019-06-17T09:30:05Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/registry.go:98" name=serviceaccount
time="2019-06-17T09:30:05Z" level=info msg="registering plugin" command=/velero kind=BlockStore logSource="pkg/plugin/registry.go:98" name=aws
time="2019-06-17T09:30:05Z" level=info msg="registering plugin" command=/velero kind=BlockStore logSource="pkg/plugin/registry.go:98" name=azure
time="2019-06-17T09:30:05Z" level=info msg="registering plugin" command=/velero kind=BlockStore logSource="pkg/plugin/registry.go:98" name=gcp
time="2019-06-17T09:30:05Z" level=info msg="registering plugin" command=/velero kind=ObjectStore logSource="pkg/plugin/registry.go:98" name=aws
time="2019-06-17T09:30:05Z" level=info msg="registering plugin" command=/velero kind=ObjectStore logSource="pkg/plugin/registry.go:98" name=azure
time="2019-06-17T09:30:05Z" level=info msg="registering plugin" command=/velero kind=ObjectStore logSource="pkg/plugin/registry.go:98" name=gcp
time="2019-06-17T09:30:05Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/registry.go:98" name=job
time="2019-06-17T09:30:05Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/registry.go:98" name=pod
time="2019-06-17T09:30:05Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/registry.go:98" name=restic
time="2019-06-17T09:30:05Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/registry.go:98" name=service
time="2019-06-17T09:30:05Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/registry.go:98" name=serviceaccount
time="2019-06-17T09:30:06Z" level=info msg="registering plugin" command=/plugins/velero-plugins kind=RestoreItemAction logSource="pkg/plugin/registry.go:98" name=or-restore-plugin
time="2019-06-17T09:30:06Z" level=info msg="registering plugin" command=/plugins/velero-plugins kind=RestoreItemAction logSource="pkg/plugin/registry.go:98" name=si-restore-plugin
time="2019-06-17T09:30:06Z" level=info msg="Checking existence of namespace" logSource="pkg/cmd/server/server.go:322" namespace=kyma-backup
time="2019-06-17T09:30:06Z" level=info msg="Namespace exists" logSource="pkg/cmd/server/server.go:328" namespace=kyma-backup
time="2019-06-17T09:30:10Z" level=info msg="Checking existence of Velero custom resource definitions" logSource="pkg/cmd/server/server.go:357"
time="2019-06-17T09:30:10Z" level=info msg="All Velero custom resource definitions exist" logSource="pkg/cmd/server/server.go:391"
time="2019-06-17T09:30:10Z" level=info msg="Checking that all backup storage locations are valid" logSource="pkg/cmd/server/server.go:398"
```
We can see that our plugins are loaded: 
```
time="2019-06-17T09:30:06Z" level=info msg="registering plugin" command=/plugins/velero-plugins kind=RestoreItemAction logSource="pkg/plugin/registry.go:98" name=or-restore-plugin
time="2019-06-17T09:30:06Z" level=info msg="registering plugin" command=/plugins/velero-plugins kind=RestoreItemAction logSource="pkg/plugin/registry.go:98" name=si-restore-plugin
```
